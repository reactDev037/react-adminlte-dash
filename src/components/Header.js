/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import styled from 'styled-components';

import Navbar from './Navbar/Navbar';
import NavbarLogo from './Navbar/NavbarLogo';
import NavbarUserMenu from './Navbar/NavbarUserMenu';
import NavbarItem from './Navbar/NavbarItem';

const StyledHeader = styled.header`
  /* clearfix */
  &:before, &:after {
    display: table;
    content: " ";
    -webkit-box-sizing: border-box;
    -moz-box-sizing: border-box;
    box-sizing: border-box;
  }
  &:after {
    clear: both;
  }

  position: ${props => (props.fixed ? 'fixed' : 'relative')};
  width: ${props => (props.boxed ? '1024px' : '100%')};
  max-height: 100px;
  z-index: 1030;
`;

const Header = props => (
  <StyledHeader fixed={props.fixed} >
    <NavbarLogo
      collapse={props.sidebarCollapse}
      sidebarMini={props.sidebarMini}
      href="#"
      logoLg={props.logoLg}
      logoSm={props.logoSm}
    />
    <Navbar
      toggle={props.sidebarToggle}
      collapse={props.sidebarCollapse}
      sidebarMini={props.sidebarMini}
    >
      {props.children}
    </Navbar>
  </StyledHeader>
);

Header.propTypes = {
  children: React.PropTypes.node,
  fixed: React.PropTypes.bool,
  logoLg: React.PropTypes.element,
  logoSm: React.PropTypes.element,
  sidebarMini: React.PropTypes.bool,
  sidebarCollapse: React.PropTypes.bool,
  sidebarToggle: React.PropTypes.func.isRequired,
};

Header.defaultProps = {
  fixed: false,
  sidebarMini: false,
  sidebarCollapse: false,
  logoLg: <span><b>Admin</b>LTE</span>,
  logoSm: <span><b>A</b>LT</span>,
};

Header.UserMenu = NavbarUserMenu;
Header.Item = NavbarItem;

export default Header;
