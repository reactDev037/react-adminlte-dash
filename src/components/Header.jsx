import React from 'react';

import HeaderLogo from './HeaderLogo';
import HeaderMenu from './HeaderMenu';

const Header = ({ children, logo, sidebarToggle }) => (
  <header className="main-header">

    {/* Logo */}
    {logo}

    {/* Header Navbar: style can be found in header.less */}
    <nav className="navbar navbar-static-top">
      {/* Sidebar toggle button*/}
      <button
        className="sidebar-toggle"
        style={{ border: 'none', outline: 'none' }}
        onClick={() => sidebarToggle()}
      >
        <span className="sr-only">Toggle navigation</span>
      </button>

      {/* Navbar Right Menu */}
      <div className="navbar-custom-menu">
        <ul className="nav navbar-nav">
          {children}
        </ul>
      </div>

    </nav>
  </header>
);

Header.propTypes = {
  children: React.PropTypes.node,
  logo: React.PropTypes.element,
  sidebarToggle: React.PropTypes.func.isRequired,
};

Header.defaultProps = {
  logo: <HeaderLogo />,
};

Header.Logo = HeaderLogo;
Header.Menu = HeaderMenu;

export default Header;
