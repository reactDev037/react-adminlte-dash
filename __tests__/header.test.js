/* eslint-env jest */
/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import { ThemeProvider } from 'styled-components';
import renderer from 'react-test-renderer';

import red from '../src/styles/skin-red';
import Header from '../src/components/Header';

describe('<Header />', () => {
  it('renders correctly with default options', () => {
    const component = renderer.create(<Header sidebarToggle={() => null} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with theme', () => {
    const component = renderer.create(<ThemeProvider theme={red}>
      <Header sidebarToggle={() => null} />
    </ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with fixed option', () => {
    const component = renderer.create(<Header fixed sidebarToggle={() => null} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with collapsed option', () => {
    const component = renderer.create(<Header sidebarCollapse sidebarToggle={() => null} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with fixed and collapsed options', () => {
    const component = renderer.create(
      <Header fixed sidebarCollapse sidebarToggle={() => null} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with given large logo', () => {
    const component = renderer.create(
      <Header
        logoLg={<span>Large</span>}
        sidebarToggle={() => null}
      />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with given small logo', () => {
    const component = renderer.create(
      <Header
        logoSm={<span>Small</span>}
        sidebarToggle={() => null}
      />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
