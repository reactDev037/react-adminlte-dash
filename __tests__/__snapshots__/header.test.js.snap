exports[`<Header /> renders correctly with collapsed option 1`] = `
<header
  className="Sdzae">
  <a
    className="bYXibN"
    href="#">
    <span
      className="ibQovi"
      href="#">
      <span>
        <b>
          Admin
        </b>
        LTE
      </span>
    </span>
    <span
      className="hboVZB"
      href="#">
      <span>
        <b>
          A
        </b>
        LT
      </span>
    </span>
  </a>
  <nav
    className="kHaBoX">
    <button
      className="fa fa-bars iXEfD"
      onClick={[Function]} />
    <div
      className="cTsgzF">
      <ul
        className="bAXwxe" />
    </div>
  </nav>
</header>
`;

exports[`<Header /> renders correctly with default options 1`] = `
<header
  className="Sdzae">
  <a
    className="bYXibN"
    href="#">
    <span
      className="ibQovi"
      href="#">
      <span>
        <b>
          Admin
        </b>
        LTE
      </span>
    </span>
    <span
      className="hboVZB"
      href="#">
      <span>
        <b>
          A
        </b>
        LT
      </span>
    </span>
  </a>
  <nav
    className="kHaBoX">
    <button
      className="fa fa-bars iXEfD"
      onClick={[Function]} />
    <div
      className="cTsgzF">
      <ul
        className="bAXwxe" />
    </div>
  </nav>
</header>
`;

exports[`<Header /> renders correctly with fixed and collapsed options 1`] = `
<header
  className="glmcVm">
  <a
    className="bYXibN"
    href="#">
    <span
      className="ibQovi"
      href="#">
      <span>
        <b>
          Admin
        </b>
        LTE
      </span>
    </span>
    <span
      className="hboVZB"
      href="#">
      <span>
        <b>
          A
        </b>
        LT
      </span>
    </span>
  </a>
  <nav
    className="kHaBoX">
    <button
      className="fa fa-bars iXEfD"
      onClick={[Function]} />
    <div
      className="cTsgzF">
      <ul
        className="bAXwxe" />
    </div>
  </nav>
</header>
`;

exports[`<Header /> renders correctly with fixed option 1`] = `
<header
  className="glmcVm">
  <a
    className="bYXibN"
    href="#">
    <span
      className="ibQovi"
      href="#">
      <span>
        <b>
          Admin
        </b>
        LTE
      </span>
    </span>
    <span
      className="hboVZB"
      href="#">
      <span>
        <b>
          A
        </b>
        LT
      </span>
    </span>
  </a>
  <nav
    className="kHaBoX">
    <button
      className="fa fa-bars iXEfD"
      onClick={[Function]} />
    <div
      className="cTsgzF">
      <ul
        className="bAXwxe" />
    </div>
  </nav>
</header>
`;

exports[`<Header /> renders correctly with given large logo 1`] = `
<header
  className="Sdzae">
  <a
    className="bYXibN"
    href="#">
    <span
      className="ibQovi"
      href="#">
      <span>
        Large
      </span>
    </span>
    <span
      className="hboVZB"
      href="#">
      <span>
        <b>
          A
        </b>
        LT
      </span>
    </span>
  </a>
  <nav
    className="kHaBoX">
    <button
      className="fa fa-bars iXEfD"
      onClick={[Function]} />
    <div
      className="cTsgzF">
      <ul
        className="bAXwxe" />
    </div>
  </nav>
</header>
`;

exports[`<Header /> renders correctly with given small logo 1`] = `
<header
  className="Sdzae">
  <a
    className="bYXibN"
    href="#">
    <span
      className="ibQovi"
      href="#">
      <span>
        <b>
          Admin
        </b>
        LTE
      </span>
    </span>
    <span
      className="hboVZB"
      href="#">
      <span>
        Small
      </span>
    </span>
  </a>
  <nav
    className="kHaBoX">
    <button
      className="fa fa-bars iXEfD"
      onClick={[Function]} />
    <div
      className="cTsgzF">
      <ul
        className="bAXwxe" />
    </div>
  </nav>
</header>
`;

exports[`<Header /> renders correctly with theme 1`] = `
<header
  className="Sdzae">
  <a
    className="ZVbxS"
    href="#">
    <span
      className="ibQovi"
      href="#">
      <span>
        <b>
          Admin
        </b>
        LTE
      </span>
    </span>
    <span
      className="hboVZB"
      href="#">
      <span>
        <b>
          A
        </b>
        LT
      </span>
    </span>
  </a>
  <nav
    className="rOyox">
    <button
      className="fa fa-bars jatJFQ"
      onClick={[Function]} />
    <div
      className="cTsgzF">
      <ul
        className="bAXwxe" />
    </div>
  </nav>
</header>
`;
